_posts/2019-05-10-continuous-deployment-the-first-step-on-the-road-to-recovery.markdown

```diff
@@ -0,0 +1,63 @@
+
+Teams choose git flow for well intentioned reasons.  Faith in the codebase is low, caused by insufficient feedback on changes provided by a comprehensive automated test suite.  This leads to regular issues being found, further reducing faith.  Teams want control over feature releases: when they are confident of the quality. Batching and branching are an attempt to provide the stability to regain control.  However, it [has been shown](https://continuousdelivery.com/evidence-case-studies/) that these practices inhibit, rather than enable, flow.
+
+# Releasing the pressure
+
+A good practice for for [The First Way](https://itrevolution.com/the-three-ways-principles-underpinning-devops/) is [one piece flow](https://www.kaizenworld.com/kaizen/one-piece-flow.html).  Work on single increments of value, delivering them as soon as possible.  
+
+As Behaviour Driven Development practitioners, we like to [_start with the end in mind_](https://cucumber.io/blog/cukeup-au-2015-videos/#sharon-robson).  We consider the goal we are trying to achieve, and work backwards to how we can achieve it.  Many teams consider one piece flow as the final step in their improvement of flow.  However, many teams never get there, nor make the changes necessary in order to deliver it.  Manual approval gates are never removed; confidence in the code remains low; automated test coverage remains poor; big batch releases are common.
+
+One piece flow - a simple idea with a simple implementation: continuous deployment. Every commit flows through your continuous delivery pipeline and, assuming all conditions pass, makes its way to production.  The idea and the implementation are simple, but there are a number of fundamental principles that you must adhere to:
+ - You have a single, consistent pipeline flow of changes into production
+ - You must have confidence in your pipeline to publish changes to the next stage
+ - In order to prevent bottlenecks, you need a highliy automated pipeline
+ - In order to prevent bottlenecks, you need a high level of automated test coverage.
+ 
+Rather than making it an aspiration, we have found that starting with continuous deployment is the first step to maximising flow.  Fundamental to that approach is [trunk based development](https://trunkbaseddevelopment.com/).  Good practice, such as branch by abstraction and feature toggling occur as a result of the move to trunk based development in order to effectively manage complexity and maintain flexibility.  System architecture becomes simple, flexible and decoupled, driven by the need to reduce risk.
+
+In trunk based development, the team works from a single master branch, with short lived feature branches for managing review and CI cycles:
+[!Trunk Based Development diagram](../img/tbd.png)**
```

Comment:
Again, ! outside the [] and reason for ** after the image url?

Reply:
updated.